examples/java/spring-boot/src/main/java/uk/gov/api/springboot/config/CorrelationIdFilter.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.api.springboot.config;
+
+import java.io.IOException;
+import java.util.UUID;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import me.jvt.uuid.Patterns;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+public class CorrelationIdFilter extends OncePerRequestFilter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CorrelationIdFilter.class);
+  private static final String CORRELATION_ID_HEADER = "correlation-id";
+
+  private final ErrorResponseDecorator errorResponseDecorator;
+  private final MdcFacade mdcFacade;
+
+  public CorrelationIdFilter(ErrorResponseDecorator errorResponseDecorator, MdcFacade mdcFacade) {
+    this.errorResponseDecorator = errorResponseDecorator;
+    this.mdcFacade = mdcFacade;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      @NonNull HttpServletRequest request,
+      @NonNull HttpServletResponse response,
+      @NonNull FilterChain filterChain)
+      throws ServletException, IOException {
+    String correlationId = retrieveOrGenerate(request);
+    if (!correlationId.matches(Patterns.UUID_STRING)) {
+      mdcFacade.put(MdcFacade.CORRELATION_ID, correlationId);
```

Comment:
We don't want to put an **invalid** correlation ID in there, as it could i.e. be a malicious string
```suggestion
```